File: 백준/Gold/14719. 빗물/빗물.java
import java.util.*;

public class Main {
    public static void main(String[] args) {
        Scanner in = new Scanner(System.in);
        int a = in.nextInt();
        int b = in.nextInt();
        int [][]num = new int[a+1][b+1];
        for(int i=1;i<=b;i++){
            int c = in.nextInt();
            for(int j=a;j>a-c;j--){
                num[j][i]=1;
            }
        }
        boolean flag = false;
        int count = 0;
        int sum=0;
        for(int i=1;i<=a;i++){
            flag = false;
            count = 0;
            for(int j=1;j<=b;j++){
                if(num[i][j]==1){
                    flag = true;
                    sum+=count;
                    count=0;
                }
                if(flag&&num[i][j]==0){
                    count++;
                }
            }
        }
        System.out.println(sum);
    }
}

File: 백준/Gold/14719. 빗물/빗물.js
const fs = require("fs");  //입력을 받아 처리하는 FileSystem 모듈 불러오기
const input = fs.readFileSync("/dev/stdin").toString().split("\n");
const arr = input[0].split(" ").map(Number); //세로길이, 가로길이
const arr2 = input[1].split(" ").map(Number); //블록이 쌓인 높이
let num = [];
for(let i=0;i<arr[0];i++){ //이차원배열 생성(모두 0으로 저장)
  let numm = [];
  for(let j=0;j<arr[1];j++){
    numm.push(0);
  }
  num.push(numm);
}

//ex) 세로 4, 가로 4, 블록의 높이가 3 0 1 4인 경우
// 0 0 0 1
// 1 0 0 1
// 1 0 0 1
// 1 0 1 1
//위와 같은 형태로 이차원 배열을 바꿔줌

for(let i=0;i<arr[1];i++){ 
  for(let j=arr[0]-1;j>=arr[0]-arr2[i];j--){
    num[j][i]=1;
  }
}
let flag = false; //쌓여 있는 블록이 있는지 확인
let count = 0; //고일 수 있는 물
let sum = 0; //최종으로 고인 물
for(let i=0;i<arr[0];i++){ //이차원 배열의 맨 윗줄부터 탐색
  flag = false; //초기화
  count = 0; //초기화
  for(let j=0;j<arr[1];j++){
    if(num[i][j]==1){ //블록이 있는 경우
      flag = true; //블록 o
      sum += count; //고일 수 있는 물 저장
      count=0; //초기화
    }
    if(flag&&num[i][j]==0){ //왼쪽에 블록이 있고 물이 고일 수 있는 공간이 있는 경우
      count++; //고일 수 있는 물 증가
    }
  }
}
console.log(sum);